### PR TITLE
Ma incident uniqness

### PR DESCRIPTION
**Pull request by money**
https://github.com/bayesimpact/bridge-uof/issues/10

### CODE REVIEW

File: app/models/general_info.rb [hunk 1]
```diff
@@ -44,6 +43,10 @@ class GeneralInfo
             numericality: { only_integer: true, greater_than: 0, less_than_or_equal_to: 20,
                             message: "should be a positive number (1-20)" }
   validates :ori, ori: true
+  validates :address, presence: true,
```
Comment (wetchler) [inline app/models/general_info.rb:46]: Also should take city/county/state into account. It's pretty realistic for there to be a "1 Main St" in several places, and they both might have an incident at the same time.

It's probably better to do this with a separate validator altogether. See http://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations

We have a few custom validators already in app/validators -- though they are all specific to a field. This one should be a whole-record validator. The link above should be helpful.

Comment (wetchler) [reply app/models/general_info.rb:46]: You'll have to query the database to see if there is already an incident matching all of these fields, and if so, raise an error. See app/validators/uniqueness_validator.rb for the idea.

File: app/models/general_info.rb [hunk 2]
```diff
@@ -44,6 +43,10 @@ class GeneralInfo
...
+                      uniqueness: { scope: [:incident_date_str,
+                                            :incident_time_str, :ori],
```
Comment (AlexNisnevich) [inline app/models/general_info.rb:48]: totally minor point, but I would put `[:incident_date_str, :incident_time_str, :ori]` on a single line.

File: app/models/general_info.rb [hunk 3]
```diff
@@ -44,6 +43,10 @@ class GeneralInfo
...
+                                    message: 'has an Incident reported with the same date and time' }
```
Comment (AlexNisnevich) [inline app/models/general_info.rb:49]: I'm wondering what the best message to put here is, to make it as clear as possible to the end user what the problem is.

Maybe something like `'has an existing incident reported for the same date and time - are you sure you didn't fill out this incident report already?'`

Ideally, if there's already an incident filled out by a different user, it should probably tell the current user who that is (e.g. "There's already an incident report filled out for this incident by Bob") - how hard do you think that would be to implement?

**Review by AlexNisnevich — CHANGES_REQUESTED** [review]
See comments. Looks good overall once tests pass.

**Pull request closed without merging**